### PR TITLE
Fix case where in memory model is not up to date with db

### PR DESCRIPTION
**Pull request by nicpottier**
Fixes: https://app.getsentry.com/nyaruka/rapidpro/issues/117827814/

### CODE REVIEW

**Comment by nicpottier** [comment]
plz review

File: temba/flows/tests.py [hunk 1]
```diff
@@ -2773,10 +2767,31 @@ def test_delete(self):
...
+        update_url = reverse('flows.flowlabel_update', args=[label_one.pk])
+
+        # not logged in, no dice
+        response = self.client.get(update_url)
+        self.assertLoginRedirect(response)
+
+        # login
+        self.login(self.admin)
+        response = self.client.get(update_url)
+
```
Comment (norkans7) [inline temba/flows/tests.py]: no assert here?

Comment (nicpottier) [inline temba/flows/tests.py]: I'm only getting it to get the initial form data, so not really needed.

Comment (norkans7) [inline temba/flows/tests.py]: yes, may be it is not needed but I think we need to see this is different to the login page this time

Comment (nicpottier) [inline temba/flows/tests.py]: Assertions below would fail in that case.

Comment (norkans7) [inline temba/flows/tests.py]: ok

**Comment by norkans7** [comment]
Looks good

File: temba/flows/models.py [hunk 2]
```diff
@@ -3680,8 +3680,7 @@ def add_message(self, msg):
 
         # if message is from contact, mark run as responded
         if not self.run.responded and msg.direction == INCOMING:
-            self.run.responded = True
-            self.run.save(update_fields=('responded',))
+            FlowRun.objects.filter(id=self.run.id, responded=False).update(responded=True)
```
Comment (rowanseymour) [inline temba/flows/models.py]: For completeness, shouldn't this also update the model instance?

Comment (nicpottier) [inline temba/flows/models.py]: Doh, ya, definitely, good catch.

**Pull request merged**